File: Third_time/Simplify Path.cpp
class Solution {
public:
    string simplifyPath(string path) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        vector<string> st;
        int len = path.size();
        string cur;
        for(int i = 0; i <= len; i++)
        {
            if(i == len || path[i] == '/')
            {
                if(cur.size())
                {
                    if(cur == "..")
                    {
                        if(!st.empty())
                            st.pop_back();
                    }
                    else if(!(cur == "."))
                        st.push_back(cur);
                    cur.clear();
                }
            }
            else
            {
                cur.push_back(path[i]);
            }
        }
        string ret;
        for(int i = 0; i < st.size(); i++)
        {
            ret.push_back('/');
            ret.append(st[i]);
        }
        return ret.size()?ret:"/";
    }
};

File: JAVA/SimplifyPath.java
import java.util.Stack;

class Solution {
    public String simplifyPath(String path) {
        // Start typing your Java solution below
        // DO NOT write main() function
        String[] tokens = path.split("/");
        Stack<String> stack = new Stack<String>();
        for(int i = 0; i < tokens.length; i++)
        {
        	String str = tokens[i];
        	if(str.equals("") || str.equals("."))
        		continue;
        	if(str.equals(".."))
        	{
        		if(stack.isEmpty())
        			continue;
        		else
        			stack.pop();
        	}
        	else
        		stack.push(str);
        }
        StringBuilder ret = new StringBuilder();
        ret.append("/");
        while(stack.size() > 1)
        {
        	ret.append(stack.get(0)).append('/');
        	stack.remove(0);
        }
        if(stack.size() == 1)
        	ret.append(stack.firstElement());
        return ret.toString();
    }
}
public class SimplifyPath {

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		// TODO Auto-generated method stub
		String path = "//home/";
		String[] tokens = path.split("/");
	}

}
